apps/src/templates/census2017/CensusForm.jsx

```diff
@@ -120,29 +171,58 @@ class CensusForm extends Component {
     console.log(JSON.stringify(error, null, 2));
   }
 
-  censusFormSubmit() {
+  validateEmail() {
+    const emailError = this.state.submission.email.includes('@') ? false : true;
+    return emailError;
+  }
+
+  validateHowMuchCS() {
+    const howMuchCSError = this.state.selectedHowMuchCS.length === 0 ? true : false;
+    return howMuchCSError;
+  }
+
+  validateTopics() {
+    const topicsError = this.state.showFollowUp && this.state.selectedTopics.length === 0 ? true : false;
+    return topicsError;
+  }
+
+  validateFrequency() {
+    const frequencyError = this.state.showFollowUp && this.state.submission.followUpFrequency === '' ? true : false;
+    return frequencyError;
+  }
+
+  validateSubmission() {
     this.setState({
-      showForm: false,
-      showThankYou: true
-    });
-    $.ajax({
-      url: "/forms/Census2017",
-      type: "post",
-      dataType: "json",
-      data: $('#census-form').serialize()
-    }).done(this.processResponse).fail(this.processError);
-    event.preventDefault();
+      errors: {
+        ...this.state.errors,
+        email: this.validateEmail(),
+        howMuchCS : this.validateHowMuchCS(),
+        topics: this.validateTopics(),
+        frequency: this.validateFrequency()
+      }
+    }, this.censusFormSubmit);
   }
 
-  componentDidMount() {
-    // Move the haml-rendered DOM section inside our protected stateful div
-    $('#school-info').appendTo(ReactDOM.findDOMNode(this.refs.schoolInfo)).show();
+  censusFormSubmit() {
+    const { errors } = this.state;
+    if (!errors.email && !errors.howMuchCS && !errors.topics && !errors.frequency) {
+      this.setState({
+        showForm: false,
+        showThankYou: true,
+      });
+      $.ajax({
+        url: "/forms/Census2017",
+        type: "post",
+        dataType: "json",
+        data: $('#census-form').serialize()
+      }).done(this.processResponse).fail(this.processError);
+      event.preventDefault();
+    }
   }
 
   render() {
-    const { selected, showForm, showThankYou } = this.state;
-
-    const showFollowUp = selected.includes("twenty_hr_some_b") || selected.includes("twenty_hr_all_b") ? true : false;
+    const { showForm, showFollowUp, showThankYou, submission, selectedHowMuchCS, selectedTopics, errors } = this.state;
+    const showErrorMsg = errors.email || errors.howMuchCS || errors.topics || errors.frequency ? true : false;
```

Comment:
The `? true : false` at the end here will only apply to the last item.

Your options are either
a) Just get rid of that end bit completely. `showErrorMsg` will not explicitly be a bool, but I assume elsewhere we only care about truthy/falsey rather than true/false, so that is probably fine.
b) Constrain your entire expression to be boolean, i.e.
```js
const showErrorMsg = (errors.email || errors.howMuchCS || errors.topics || errors.frequency) ? true : false;
// shorter way that I tend to prefer
const showErrorMsg = !!(errors.email || errors.howMuchCS || errors.topics || errors.frequency);
```